modules/apps/feature-flag/feature-flag-web/src/main/resources/META-INF/resources/js/FeatureFlagToggle.tsx

```diff
@@ -30,6 +38,18 @@ const FeatureFlagToggle = ({
 	const [disabled, setDisabled] = useState(initialDisabled);
 	const [toggled, setToggled] = useState(initialToggled);
 
+	FeatureFlagToggleRegistry.registerToggle(
+		featureFlagKey,
+		(newToggled: boolean, newDisabled: boolean) => {
+			if (toggled !== newToggled) {
+				setToggled(newToggled);
+			}
+			if (disabled !== newDisabled) {
+				setDisabled(newDisabled);
+			}
+		}
+	);
```

Comment:
This is not considered good practice to update the state during the rendering lifecycle, it is very likely that React.js may throw an error in the console. Maybe you can move this to a `useEffect`.